### PR TITLE
introduced snackbar react to smart contract events and rejected transactions

### LINKED ISSUE #19: Add Snack Bar for smart contract events
A nice little snackbar indicating that the withdraw/deposit was successful (or not)

### PR DESCRIPTION
**Pull request by yvesbou**
need to implement failure snackbars

### CODE REVIEW

**Comment by yvesbou** [comment]
found a bug introduced with #17 
use to be:  `if (currentAccount === from){}`
now:  `if (currentAccount === from.toLowerCase()){}`
basically, we receive the address of current account from metamask with only lower case, while the event emits upper and lower case

**Comment by leovct** [comment]
> found a bug introduced with #17 use to be: `if (currentAccount === from){}` now: `if (currentAccount === from.toLowerCase()){}` basically, we receive the address of current account from metamask with only lower case, while the event emits upper and lower case

I'm curious, what was the error message?

File: src/App.js [hunk 1]
```diff
@@ -73,7 +82,21 @@ export default function App() {
...
         const fundContract = new ethers.Contract(CONTRACT_ADDRESS, contractABI, signer);
 
-        await fundContract.fund({ value: ethers.utils.parseEther(amountDeposit.toString())});
+        fundContract
+          .fund({ value: ethers.utils.parseEther(amountDeposit.toString())})
+          .then((tx)=>{
+            //action prior to transaction being mined
+            provider.waitForTransaction(tx.hash)
+          .then(()=>{
+             //action after transaction is mined
```
Comment (leovct) [inline src/App.js]: Can you also display the snackbar when the transaction was successfull?

File: src/App.js [hunk 2]
```diff
@@ -73,7 +82,21 @@ export default function App() {
         const signer = provider.getSigner();
         const fundContract = new ethers.Contract(CONTRACT_ADDRESS, contractABI, signer);
 
-        await fundContract.fund({ value: ethers.utils.parseEther(amountDeposit.toString())});
+        fundContract
+          .fund({ value: ethers.utils.parseEther(amountDeposit.toString())})
+          .then((tx)=>{
+            //action prior to transaction being mined
```
Comment (leovct) [inline src/App.js]: And here when the transaction was sent (but not mined yet)?

File: src/App.js [hunk 3]
```diff
@@ -90,7 +113,21 @@ export default function App() {
         const signer = provider.getSigner();
         const fundContract = new ethers.Contract(CONTRACT_ADDRESS, contractABI, signer);
 
-        fundContract.withdraw(ethers.utils.parseEther(amountWithdraw.toString()));
+        fundContract
+          .withdraw(ethers.utils.parseEther(amountWithdraw.toString()))
+          .then((tx)=>{
+            //action prior to transaction being mined
```
Comment (leovct) [inline src/App.js]: Same thing here.

File: src/App.js [hunk 4]
```diff
@@ -90,7 +113,21 @@ export default function App() {
...
+            provider.waitForTransaction(tx.hash)
+          .then(()=>{
+             //action after transaction is mined
```
Comment (leovct) [inline src/App.js]: Same thing here.

**Comment by yvesbou** [comment]
> > found a bug introduced with #17 use to be: `if (currentAccount === from){}` now: `if (currentAccount === from.toLowerCase()){}` basically, we receive the address of current account from metamask with only lower case, while the event emits upper and lower case
> 
> I'm curious, what was the error message?

Sorry not a bug (but wrong behavior), but the expression evaluated false and therefore the conditional code wasn't executed, even when it should! That's why "Your AVAX Funds" did not update quickly as the "Total Funds" in my Demo on Tuesday.

**Comment by yvesbou** [comment]
Improvements to be done before merging:
- [x] display the snackbar when the deposit was successful
- [x] display the snackbar when the withdraw was successful
- [x] display the snackbar when the deposit transaction was sent (but not mined yet)
- [x] display the snackbar when the withdraw transaction was sent (but not mined yet)

## Update
The Snackbars show up in the right order, however, they messages for "mined" and "smart contract event" come within ultra short time, such that it the "mined" message is already replaced by the "smart contract event" message, before the user would have time to read it. Tried with notistack (stack snackbars) but their messages are hard to control (too many, repeatedly triggered)

I will try tomorrow and if it doesn't work, I leave out the second of three which is mined:

 sent - (mined) - event

## Update 2
I included a new dependency called notistack, which allowed me to stack snackbars. I managed to display each snackbar only once. Queuing the state changes directly after another.
I decided to leave out the event snackbar, it was a bit unclear for me, when putting myself in the shoes of a user, why mined and then event also comes, it seemed to me a bit too much.

Regarding showing a link to snowtrace, I didn't manage to do it. I want to create a separate issue, which can be additional feature extension to this. But I want to close this PR with the current functionality.

**Review by leovct — APPROVED** [review]

**Comment by leovct** [comment]
No problem for the tx explorer link, we'll see if this is a feature we want to implement later! :)

**Pull request merged**